content/learn/book/the-game-loop/time-and-timers.md

```diff
@@ -0,0 +1,277 @@
+  - otherwise, this is used for all of your gameplay logic as well
+- fixed time
+  - all of your gameplay logic should go here if you're using a fixed timestep approach
+
+When thinking about fixed time, it's important to clearly distinguish **frames** and **ticks**.
+A "frame" is one pass of the [`Main`] schedule, and corresponds to a single rendered frames.
+A "tick" is one pass of the [`FixedMain`] schedule, and corresponds to one iteration of gameplay logic.
+The ratio between frames elapsed and ticks elapsed is not constant: they may be faster or slower, depending
+on the rendering performance and your game's simulation needs.
+
+{% callout(type="info") %}
+Simply requesting [`Time`] in your systems will get you the correct flavor 90% of the time: virtual time in the [`Main`] schedule,
+and fixed time in the [`FixedMain`] schedule.
+To request a specific variation, change the implicit generic in [`Time`] from `()`
+to [`Real`], [`Virtual`] or [`Fixed`].
+{% end %}
+
+Now that we have the required vocabulary, let's go over exactly how the fixed timestep logic works in Bevy:
+
+- each frame, during the [`RunFixedMainLoop`] schedule, we determine how many times we should run the [`FixedMain`] schedule
+  - we add the the elapsed virtual time for the frame to a running time buffer
+  - if that time buffer is greater than or equal to [`Time<Fixed>::timestep`], we run the [`FixedMain`] schedule once, and subtract that timestep from our buffer
+- when the [`FixedMain`] schedule is evaluated, the various subschedules are iterated through, running [`FixedPreUpdate`], [`FixedUpdate`] and so on in order
+- within your fixed time systems, use the exact same delta time techniques to properly account for changes in your desired timestep during development
+
+This means that we may have 0, 1, 2 or more ticks per frame,
+with our fixed timestep logic running repeatedly until it's caught back up.
+For even more detail, check out the excellent documentation on [`Fixed`].
+
+Note that Bevy's "fixed timesteps" are not the right mechanism to use for gameplay logic like "every 5 seconds update this building".
+Bevy only supports a single fixed timestep across your entire project, and its use is completely optional.
+Please see our [timers and cooldowns] section below for the tools available to model this type of behavior.
+
+[`Time<Fixed>::timestep`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Time.html#method.timestep
+[`Main`]: https://docs.rs/bevy/latest/bevy/app/struct.Main.html
+[`FixedMain`]: https://docs.rs/bevy/latest/bevy/app/struct.FixedMain.html
+[`Real`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Real.html
+[`Virtual`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Virtual.html
+[`Fixed`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Fixed.html
+[fix your timestep article]: https://gafferongames.com/post/fix_your_timestep/
+[`RunFixedMainLoop`]: https://dev-docs.bevy.org/bevy/app/struct.RunFixedMainLoop.html
```

Comment:
You probably wanted docs.rs ;)

Reply:
Depends on which version these docs are targeting. It's called [`RunFixedMainLoopSystem`](https://docs.rs/bevy/latest/bevy/app/enum.RunFixedMainLoopSystem.html) in 0.16, and `RunFixedMainLoop` on the main branch / in 0.17.

---

Side tangent, these links use `docs.rs/bevy/latest`; we'll want to switch the book's links to use a specific version like 0.16 at some point so that links aren't broken for old builds of the book, especially if we make the book versioned. Maybe we could even have a CI test that denies docs.rs links with `/latest/` to avoid links like that accidentally slipping through?